Python/import.c

```diff
@@ -2249,6 +2249,9 @@ _imp_create_dynamic_impl(PyObject *module, PyObject *spec, PyObject *file)
 
     mod = _PyImport_FindExtensionObject(name, path);
     if (mod != NULL || PyErr_Occurred()) {
+        if (PyErr_Occurred()) {
+            PyErr_Clear();
+        }
```

Comment:
This actually won't work because since you cleared out the error but are returning `NULL` you will trigger a failure in the interpreter for saying there's an exception when one isn't set.